pkg/fanal/analyzer/analyzer.go

```diff
@@ -323,8 +332,23 @@ func NewAnalyzerGroup(opts AnalyzerOptions) (AnalyzerGroup, error) {
 	group := AnalyzerGroup{
 		logger:            log.WithPrefix("analyzer"),
 		filePatterns:      make(map[Type][]*regexp.Regexp),
+		fileSizes:         make(map[Type]int64),
 		detectionPriority: opts.DetectionPriority,
 	}
+	for _, f := range opts.MaxFileSize {
+		// e.g. "jar:5mb secret:5kb"
+		s := strings.SplitN(f, separator, 2)
+		if len(s) != 2 {
+			return group, xerrors.Errorf("invalid max file size (%s) expected format: \"analyzerType:maxFileSize\" e.g. \"jar:5mb secret:5kb\"", f)
+		}
+
+		analyzerType := Type(s[0])
+		maxFileSize, err := units.FromHumanSize(s[1])
+		if err != nil {
+			return group, xerrors.Errorf("invalid file size (%s): %w", s[1], err)
+		}
+		group.fileSizes[analyzerType] = maxFileSize
+	}
```

Comment:
What if we move this logic into the `flag` package:
https://github.com/aquasecurity/trivy/blob/1aa51589050f1870035bf6bf75aee408b8f7ae4c/pkg/flag/scan_flags.go#L183-L212

PS. `opts.MaxFileSize` will be `map[Type]int64`.

Then users will get an error message if the wrong template is used immediately when launching Trivy.
@knqyf263 wdyt?

Reply:
Sounds good